### PR TITLE
Add basic API for background image

### LINKED ISSUE #522: Image as background
## Description

I'm probably being too lazy, but I haven't found any example of setting a background of `Plotter` to an image. Is it possible at all? `Plotter.set_background()` seems to work only with colours.

Thanks.

### PR DESCRIPTION
**Pull request by GuillaumeFavelier**
Following https://github.com/pyvista/pyvista/issues/522#issuecomment-583095108, this PR is an experiment for basic support of an image as background. I could notice three constraints with this implementation:

1) The background image is shared among the viewports/renderers
2) The image is not resized with the window
3) It's available only for `Plotter` at the moment

Closes #522

### CODE REVIEW

**Comment by GuillaumeFavelier** [comment]
I kept the changes 'local' to `__init__()` to avoid to break anything and make integration easier

**Comment by akaszynski** [comment]
Went ahead and changed this to `add_background_image` as a method for the `Plotter` class.

```python
import pyvista as pv
from pyvista import examples

pl = pv.Plotter(shape=(2, 2))

pl.subplot(0, 0)
pl.add_text('Earth Visible as Map')
pl.add_background_image(examples.mapfile, as_global=False)

pl.subplot(0, 1)
pl.add_text('Earth Altitude')
actor = pl.add_mesh(earth_alt, cmap='gist_earth')

pl.subplot(1, 0)
topo = examples.download_topo_land()
actor = pl.add_mesh(topo, cmap='gist_earth')
pl.add_text('Earth Land Altitude')

pl.subplot(1, 1)
pl.add_text('Earth Visible as Globe')
pl.add_mesh(examples.load_globe(), smooth_shading=True)

pl.show()
```
desktop 1_152

**Comment by akaszynski** [comment]
Ready for review @pyvista/developers

File: pyvista/plotting/plotting.py [hunk 1]
```diff
@@ -3836,3 +3834,98 @@ def plot(self, *args, **kwargs):
...
+        resize_background()
+
+        # setup autoscaling of the image
+        if auto_resize and hasattr(self, 'iren'):  # pragma: no cover
+            def on_resize(*args):
+                if self.__prior_window_size != self.window_size:
+                    self.__prior_window_size = self.window_size
+                    resize_background()
+
+            self.iren.AddObserver('ModifiedEvent', on_resize)
```
Comment (GuillaumeFavelier) [inline pyvista/plotting/plotting.py]: We should use our brand new `_add_observer` here.

Comment (akaszynski) [reply pyvista/plotting/plotting.py]: Now using it as of e1f848b307f08cdf03442cc02911ffead8a2945a.

**Comment by akaszynski** [comment]
Added 

> Now, I would like to discuss about the API around this. So first question, with viewport support and `add_background_image`, it is possible to add multiple background images but all the renderers are stored in the same `self._background_renderer` variable, what do you think about creating `self.background_renderers` and point to the current one with `self._background_renderer` (exactly like we do with `self.renderer`?

Ended up adding `self._background_renderers` to the `BasePlotter` class.  They're also cleaned up on destruction.  This allows us to add several background images in a subplot.  Also, it now works for the `BackgroundPlotter`:

image

> 
> I also think that a `remove_background_image()` would be nice to be consistent with our design. And finally, I think we should clean the memory like we do with the normal renderer. Maybe in this case, removing the image actor is enough but I wanted to point it out.  So except from `_add_observer` and my memory concern, I think the rest can be done in follow-up PRs.

This PR should take care of all those concerns and it's ready for review again.

**Comment by akaszynski** [comment]
pinging @GuillaumeFavelier for review

**Comment by GuillaumeFavelier** [comment]
Sorry for the delay @akaszynski, I'll definitely take a look later in the day :+1:

**Comment by banesullivan** [comment]
Is it possible to pass a 2D NumPy array as the background image?

Perhaps we should add a way to handle more ways of using an image in memory in addition to passing an image file path.

**Pull request merged**